### PR TITLE
Create ExecutionFlowDBManager class to migrate to use new azkaban db module

### PR DESCRIPTION
**Pull request by kunkun-tang**
The goal of this code patch is to gradually migrate to new azkaban DB module. I construct a new class `ExecutorFlowDBManager ` to store Table `execution_flow` related DB operations. Basically, I move a few methods out from `JdbcExecutorLoader` to new class and leverage  new DB API.  There are 3 benefits for the new class:

* Split #1291 into a few commits, and will make reviews easier.
* Put Query Handler class and its caller together. In this way, make code clean and easy to read.
* Also easy to manage unit Tests.

I'm considering using `ExecutorFlowDBManager` directly to replace `JdbcExecutorLoader`, when it comes to `execution_flows` related operations. However, Since some classes like `FlowRunner` has not been guificied, it is not easy to add another dependency. We will do the follow-up refactor after executor loader is completed migrated.

Evaluated via Integrate Tests.

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/executor/ExecutorFlowDBManager.java [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+  public synchronized void uploadExecutableFlow(final ExecutableFlow flow)
+      throws ExecutorManagerException {
+    final String INSERT_EXECUTABLE_FLOW = "INSERT INTO execution_flows "
+        + "(project_id, flow_id, version, status, submit_time, submit_user, update_time) "
+        + "values (?,?,?,?,?,?,?)";
+    final long submitTime = System.currentTimeMillis();
+
+    final SQLTransaction<Long> insertAndGetLastID = transOperator -> {
+      transOperator.update(INSERT_EXECUTABLE_FLOW, flow.getProjectId(),
```
Comment (jamiesjc) [inline azkaban-common/src/main/java/azkaban/executor/ExecutorFlowDBManager.java]: What would happen if update throws SQLException here?

Comment (jamiesjc) [inline azkaban-common/src/main/java/azkaban/executor/ExecutorFlowDBManager.java]: Never mind. Just realized it was handled below.

File: azkaban-common/src/test/java/azkaban/executor/ExecutorFlowDBManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    assertThat(flowList2.size()).isEqualTo(1);
+
+    final ExecutableFlow fetchFlow =
+        this.executorFlowDBManager.fetchExecutableFlow(flow.getExecutionId());
+    assertTwoFlowSame(flowList1.get(0), flowList2.get(0));
+    assertTwoFlowSame(flowList1.get(0), fetchFlow);
+  }
+
+  private void assertTwoFlowSame(final ExecutableFlow flow1, final ExecutableFlow flow2) {
+    assertThat(flow1.getExecutionId()).isEqualTo(flow2.getExecutionId());
```
Comment (jamiesjc) [inline azkaban-common/src/test/java/azkaban/executor/ExecutorFlowDBManagerTest.java]: Consider also comparing flow status?

Comment (kunkun-tang) [inline azkaban-common/src/test/java/azkaban/executor/ExecutorFlowDBManagerTest.java]: Good suggestion! Adding.

**Comment by jamiesjc** [comment]
Can you add more details regarding your plans on the future refactor work?
e.g. Do you plan to move all execution_flows table related DB operations to this ExecutionFlowDBManager? Also do you plan to move some DB operations which involve multiple DB tables, e.g. FetchActiveExecutableFlows, to separate classes?

File: azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDBManager.java [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+      final List<ExecutableFlow> execFlows = new ArrayList<>();
+      do {
+        final int id = rs.getInt(1);
+        final int encodingType = rs.getInt(2);
+        final byte[] data = rs.getBytes(3);
+
+        if (data != null) {
+          final EncodingType encType = EncodingType.fromInteger(encodingType);
+          final Object flowObj;
+          try {
```
Comment (jamiesjc) [inline azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDBManager.java]: Just add a note here:
This part of the code is duplicate in multiple places in the code base now. When you do future refactoring, can you also improve it?

Comment (kunkun-tang) [inline azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDBManager.java]: I added a TODO comment to fix this.

File: azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDBManager.java [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.inject.Inject;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.apache.log4j.Logger;
+
+@Singleton
+public class ExecutionFlowDBManager {
```
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDBManager.java]: According to the Google style guide, the name should be ExecutionFlowDbManager.
see
https://google.github.io/styleguide/javaguide.html#s5.3-camel-case

**Pull request merged**